src/framework/auth/services/token/token-storage.ts

```diff
@@ -27,27 +27,28 @@ export abstract class NbTokenStorage {
  *
  */
 @Injectable()
-export class NbTokenLocalStorage implements NbTokenStorage {
+export class NbTokenLocalStorage extends NbTokenStorage {
 
   protected key = 'auth_app_token';
 
-  constructor(@Inject(NB_AUTH_TOKEN_CLASS) protected tokenClass: NbTokenClass) {
+  constructor(private tokenPacker: NbAuthTokenPacker) {
+    super();
   }
 
   /**
    * Returns token from localStorage
    * @returns {NbAuthToken}
    */
   get(): NbAuthToken {
-    return nbCreateToken(this.tokenClass, localStorage.getItem(this.key));
+    return this.tokenPacker.unpack(localStorage.getItem(this.key));
```

Comment:
Maybe, it would be cleaner to not call one function in another? And do this operation in two lines?